### PR TITLE
[PS2] Initial `PS2` Port

### LINKED ISSUE #5723: Infinite Loop in devices without native Atomic support
Hello guys,
I'm starting a Playstation 2 port of SDL, so far I have made some progress already, however running the tests I have noticed an infinite loop.

Having in mind that PS2 doesn't have native atomic support, we have the following scenario:

1. Function `SDL_CreateMutex` calling `SDL_Malloc` https://github.com/libsdl-org/SDL/blob/main/src/thread/generic/SDL_sysmutex.c#L43
2. Function `SDL_malloc` calling `SDL_AtomicIncRef` https://github.com/libsdl-org/SDL/blob/main/src/stdlib/SDL_malloc.c#L5390
3. `#define SDL_AtomicIncRef(a)    SDL_AtomicAdd(a, 1)` https://github.com/libsdl-org/SDL/blob/main/include/SDL_atomic.h#L321
4. Function `SDL_AtomicAdd` calling `SDL_AtomicCAS` https://github.com/libsdl-org/SDL/blob/main/src/atomic/SDL_atomic.c#L257
5. Function `SDL_AtomicCAS` calling `enterLock` https://github.com/libsdl-org/SDL/blob/main/src/atomic/SDL_atomic.c#L145
6. Function `enterLock` calling `SDL_AtomicLock` https://github.com/libsdl-org/SDL/blob/main/src/atomic/SDL_atomic.c#L114
7. Function `SDL_AtomicLock` calling `SDL_AtomicTryLock` https://github.com/libsdl-org/SDL/blob/main/src/atomic/SDL_spinlock.c#L165
8. Function `SDL_AtomicTryLock` calling `SDL_CreateMutex` https://github.com/libsdl-org/SDL/blob/main/src/atomic/SDL_spinlock.c#L63 

What is your suggestion to follow?

I was thinking of implementing my own "atomic" support, by disabling interruptions, then performing the operation, and finally, enabling interruption backs.
https://github.com/libsdl-org/SDL/blob/main/src/atomic/SDL_atomic.c#L236

### PR DESCRIPTION
**Pull request by fjtrujy**
## Description
This PR is the first one of a collection, with the main objective to have a usable `SDL2` Port for Playstation 2.

This PR contains:
- Atomic support
- `SDL_Log` driver
- `SDL CPU info`
- `SDL Systimer` driver
- `SDL_Semaphore` driver
-  `SDL_Mutex` driver
- `SDL_Thread` driver

Additionally, I have also done:
- `cmake` support for library
- `cmake` support for tests
- Implemented the CI/CD
- Other minor changes.

Closes #5723 

## Testing
I have performed most of the test that requires the drivers that I have implemented.
It can be tested either with real PS2 hardware or with the `PCSX2` emulator.

If you want to run it in the emulator, first be sure you have the `filesystem` enabled.
Screenshot 2022-06-15 at 23 10 09
Then just load the desired `.elf` file.
Screenshot 2022-06-15 at 23 11 17

Finally, I have uploaded to the workflow `CI/CD`, all the generated `elf` samples.

## What's pending?
For the rest of the pending drivers, I would like to create them, in separate PRs. 

Special mention to @davidgfnet for his continuous help and dedication.

Cheers.

### CODE REVIEW

File: docs/README-ps2.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+PS2
+======
+SDL2 port for the Sony Playstation 2 contributed by:
+- Francisco Javier Trujillo Mata
+
+
+Credit to
+   - The guys that ported SDL to PSP & Vita because I'm taking them as renference.
```
Comment (slouken) [inline docs/README-ps2.md:8]: type: reference

File: src/dynapi/SDL_dynapi.h [hunk 2]
```diff
@@ -51,8 +51,12 @@
 #define SDL_DYNAMIC_API 0
 #elif defined(SDL_BUILDING_WINRT) && SDL_BUILDING_WINRT  /* probably not useful on WinRT, given current .dll loading restrictions */
 #define SDL_DYNAMIC_API 0
+#elif defined(__PS2__) && __PS2__
+#define SDL_DYNAMIC_API 0
 #elif defined(__PSP__) && __PSP__
 #define SDL_DYNAMIC_API 0
+#elif defined(_EE) /* probably not useful on PS2 */
```
Comment (slouken) [inline src/dynapi/SDL_dynapi.h:58]: What is _EE here?

Comment (madebr) [reply src/dynapi/SDL_dynapi.h:58]: [Emotion Engine](https://en.wikipedia.org/wiki/Emotion_Engine), the CPU used by Sony in PS2.

File: src/filesystem/ps2/SDL_sysfilesystem.c [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    SDL_snprintf(retval, len, "%s%s/%s/", base, org, app);
+  } else {
+    SDL_snprintf(retval, len, "%s%s/", base, app);
+  }
+  free(base);
+
+  printf("Creating folder %s\n", retval);
+  mkdir(retval, 0x0755);
+  
+  while(1) {}
```
Comment (eloj) [inline src/filesystem/ps2/SDL_sysfilesystem.c:76]: Debug vestige?!

File: src/dynapi/SDL_dynapi.h (hunk 2 above)
Comment (madebr) [reply src/dynapi/SDL_dynapi.h:58]: EE is the cpu, and DVP is the vector processor.
ps2dev (=the organization that maintains the ps2 toolchain) provides a C compiler that targets EE and an assembler for DVP.

Comment (slouken) [reply src/dynapi/SDL_dynapi.h:58]: Right, but we've already checked __PS2__ above... ?

Comment (fjtrujy) [reply src/dynapi/SDL_dynapi.h:58]: yes, I need to remove it and use just `__PS2__`

File: src/filesystem/ps2/SDL_sysfilesystem.c (hunk 3 above)
Comment (fjtrujy) [reply src/filesystem/ps2/SDL_sysfilesystem.c]: Thanks, removed it

File: docs/README-ps2.md (hunk 1 above)
Comment (fjtrujy) [reply docs/README-ps2.md]: Fixed! Thanks

**Pull request merged**